tests/storage/datavolume.go

```diff
@@ -1094,6 +1096,66 @@ var _ = SIGDescribe("DataVolume Integration", func() {
 			Entry("[test_id:5896]with thick provision false, fstrim will make the image smaller", addThickProvisionedFalseAnnotation, true),
 		)
 	})
+
+	Context("VirtualMachinePreference", func() {
+		var vm *v1.VirtualMachine
+		var virtualMachinePreference *instanceType.VirtualMachinePreference
+		var foundSC bool
+		var pvcName string
+
+		BeforeEach(func() {
+			vm, foundSC = tests.NewRandomVMWithDataVolume(cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskAlpine), util.NamespaceTestDefault)
+			if !foundSC {
+				Skip("Skip test when Filesystem storage is not present")
+			}
+
+			// Remove storage class name
+			vm.Spec.DataVolumeTemplates[0].Spec.PVC.StorageClassName = nil
+
+			// Get PVC name
+			pvcName = vm.Spec.DataVolumeTemplates[0].Name
+
+			// Create VirtualMachinePreference
+			preferredVolumes := &instanceType.VolumePreferences{
+				PreferredStorageClassName: "test-storage-class",
+			}
+			virtualMachinePreference = &instanceType.VirtualMachinePreference{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "virtualmachinepreference-test",
+					Namespace: vm.Namespace,
+				},
+				Spec: instanceType.VirtualMachinePreferenceSpec{
+					Volumes: preferredVolumes,
+				},
+			}
+
+			_, err := virtClient.VirtualMachinePreference(util.NamespaceTestDefault).Create(context.Background(), virtualMachinePreference, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			Expect(virtClient.VirtualMachine(vm.Namespace).Delete(vm.Name, &metav1.DeleteOptions{})).To(Succeed())
+			Expect(virtClient.VirtualMachinePreference(virtualMachinePreference.Namespace).Delete(context.Background(), virtualMachinePreference.Name, metav1.DeleteOptions{})).To(Succeed())
+		})
+
+		It("should create PVC with VirtualMachinePreference storage class", func() {
+			preferenceMatcher := &v1.PreferenceMatcher{
+				Name: virtualMachinePreference.Name,
+				Kind: "VirtualMachinePreference",
+			}
+			vm.Spec.Preference = preferenceMatcher
+
+			// Created VM with VirtualMachinePreference
+			vm, err = virtClient.VirtualMachine(util.NamespaceTestDefault).Create(vm)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(ThisPVCWith(vm.Namespace, pvcName), 160).Should(Exist())
+			pvc, err := virtClient.CoreV1().PersistentVolumeClaims(vm.Namespace).Get(context.Background(), pvcName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(pvc.Spec.StorageClassName).To(Equal(&virtualMachinePreference.Spec.Volumes.PreferredStorageClassName))
+		})
```

Comment:
Nice, I think we are missing coverage when the user has provided a `StorageClassName` right?